MUXSDKStats/MUXSDKStats/MUXSDKStats.h

```diff
@@ -178,6 +178,17 @@ FOUNDATION_EXPORT
  @discussion  Use this method to signal that the player is now playing a new video. The player name provided must been passed as the name in a monitorPlayer:withPlayerName:andConfig: call. The config provided should match the specifications in the Mux docs at https://docs.mux.com and should include all desired keys, not just those keys that are specific to this video. If the name of the player provided was not previously initialized, an exception will be raised.
 
  */
++ (void)setAutomaticVideoChange:(nonnull NSString *)name enabled:(nullable Boolean *)enabled;
+
+/*!
+ @method      setAutomaticVideoChange:forPlayer:enabled
+ @abstract    Allows default videochange functionality to be disabled.
+ @param       name The name of the player to update
+ @param       enabled Whether to enabled or disable the automatic videochange functionality
+ @discussion  Use this method to disable built in videochange calls when using AVQueuePlayer.. The player name provided must been passed as the name in a monitorPlayer:withPlayerName:andConfig: call. The config provided should match the specifications in the Mux docs at https://docs.mux.com and should set the enabled value to false. The default setting is true. If the name of the player provided was not previously initialized, an exception will be raised.
```

Comment:
you have an extra `.` in here on the first line.

Reply:
Also, given how this is written, I don't think it raises an exception if the player wasn't initialized, I think it just straight up no-ops. Right?

Reply:
👍  i'll clean the extra `.`. On the exception, this was a question I had myself - I guess a no-op is what we'd prefer?

Reply:
I'd just copy what the error tracking flag does (and if the docs are wrong on the other one, update the docs there too)

Reply:
I think having a new public method like `setAutomaticVideoChange:name:enabled` is fine but we should also have a discussion about how we want various options/flags that modify our default behaviors to be represented in this SDK. 

We initially added the automatic error tracking flag by adding additional monitor methods:
```
+ (MUXSDKPlayerBinding *_Nullable)monitorAVPlayerViewController:(nonnull AVPlayerViewController *)player withPlayerName:(nonnull NSString *)name playerData:(nonnull MUXSDKCustomerPlayerData *)playerData videoData:(nullable MUXSDKCustomerVideoData *)videoData automaticErrorTracking:(BOOL)automaticErrorTracking;
```

Then later, when we added `viewData`, we needed to add additional public methods that also had this error tracking flag:
```
+ (MUXSDKPlayerBinding *_Nullable)monitorAVPlayerViewController:(nonnull AVPlayerViewController *)player withPlayerName:(nonnull NSString *)name playerData:(nonnull MUXSDKCustomerPlayerData *)playerData videoData:(nullable MUXSDKCustomerVideoData *)videoData viewData: (nullable MUXSDKCustomerViewData *) viewData automaticErrorTracking:(BOOL)automaticErrorTracking;
```

This approach will start to get unscalable as we have more options because we will end up with a large number of overloaded ways to call `monitor`.

Our other option is to do what you've done here which is to add a special setter. But this approach also can also get difficult to for end users of this SDK as more options are added because we will have a setter for each one and internally we will also have to remember to implement consistent behavior if the player isn't initialized etc. 

The most idiomatic way to represent options in Obj-C/Swift APIs is to use [NS_OPTIONS macros in Obj-C](https://developer.apple.com/library/archive/releasenotes/ObjectiveC/ModernizationObjC/AdoptingModernObjective-C/AdoptingModernObjective-C.html#//apple_ref/doc/uid/TP40014150-CH1-SW6) which get translated to the [OptionSet in Swift](https://nshipster.com/optionset/). 

Apple uses this approach heavily so it will be familiar to iOS developers.
```
typedef NS_OPTIONS(NSUInteger, MUXSDKOptions) {
    MUXSDKOptionsNone                    = 0,
    MUXSDKOptionsAutomaticErrorTracking  = 1 << 0,
    MUXSDKOptionsAutomaticVideoChange    = 1 << 1,
};
```
Then you can define methods like this:
```
+ (MUXSDKPlayerBinding *_Nullable)monitorAVPlayerViewController:(nonnull AVPlayerViewController *)player withPlayerName:(nonnull NSString *)name playerData:(nonnull MUXSDKCustomerPlayerData *)playerData videoData:(nullable MUXSDKCustomerVideoData *)videoData options:(MUXSDKOptions) options;
```

Check options like this:
```
(options & MUXSDKOptionsAutomaticErrorTracking) == MUXSDKOptionsAutomaticErrorTracking)
```

Set options like this:
```
options = MUXSDKOptionsAutomaticErrorTracking | MUXSDKOptionsAutomaticVideoChange
```

I believe if we did this and moved error tracking flag to this approach it would be a major version bump so I'm not suggesting we do that right now, just wanted to talk about it.

Reply:
Yes, I really like this options approach, honestly. This is going to get out of hand, quickly. Part of why we added automaticErrorTracking to the constructor, though, was namely because we needed to be able to disable that for errors that happened _during setup_. I think I'd suggest going with the current proposal, but also follow up with `MUXSDKOptions` as we start to join these together.

Reply:
That sounds good to me!

Reply:
I also love this idea long term! For now, i've tidied up the description but we should revisit this for a major bump at some point. I'll write a ticket for the backlog!